src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1166,7 +1166,8 @@ private NamedTypeSymbol ConstructNamedTypeUnlessTypeArgumentOmitted(SyntaxNode t
 
             if (ShouldCheckConstraints)
             {
-                type.CheckConstraintsForNonTuple(this.Conversions, typeSyntax, typeArgumentsSyntax, this.Compilation, basesBeingResolved, diagnostics);
+                bool includeNullability = Compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticNullChecking);
+                type.CheckConstraintsForNonTuple(this.Conversions.WithNullability(includeNullability), typeSyntax, typeArgumentsSyntax, this.Compilation, basesBeingResolved, diagnostics);
```

Comment:
>CheckConstraintsForNonTuple [](start = 21, length = 27)

Why are tuples vs. non-tuples relevant here?
Do we need to test a ValueTuple type with constraints?

Reply:
This is not specific to nullable - this call is in master. And it looks like the transform to `TupleTypeSymbol` occurs below, after constraints are checked.

---
In reply to: [207374410](https://github.com/dotnet/roslyn/pull/28959#discussion_r207374410) [](ancestors = 207374410)